### ISSUE TITLE
on.(“signal”) event repeats many times

### ISSUE THREAD

**Issue by rakcode** [feross/simple-peer#710]
I'm using simple-peer to create a small video calling app using NodeJS, SocketIO and Angular

I'm trying to implement full-mesh here. I have a peer initialtor which will initiate the signal on loop.

My code is as below.

```
const peer = new SimplePeer({
    initiator: true,
    trickle: false,
    stream,
});

peer.on("signal", signal => {
    console.log(userToSignal, "signal-callerID");
    this.peerService.emit("sending signal", { userToSignal, callerID, signal })
})
```
This function will be in loop. when a new user conenct, It'll send the signal to that user, My problem is, peer.on("signal", signal => event repeats itself even if there is only 1 user conencted, its making likle 11 requests on 1 peer create. whats the issue here and how can i fix it?

I'm refering to [this code](https://github.com/coding-with-chaim/group-video-final) and trying to implement this in angular. This code is in React, there the event is working fine,

**Comment by nazar-pc** [comment]
Look into the contents of the events. It is most likely an error in signaling code that can be fixed after some debugging (#702). Similar issue was reported just over a week ago and was solved by fixing application code.

I'm closing for now since there doesn't seem anything to be done in simple-peer about this, but feel free to continue discussion. If there is an indication of an actual issue in simple-peer, we'll reopen this issue.

**Comment by etienne-martin** [comment]
Set the `trickle` option to `false` if you want to get a single 'signal' event (slower)
